<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Analyse des Interactions de protéines | Tutoriel de Bioinformatique</title>
  <meta name="description" content="Ce document est recueil qui permettra aux étudiants d’avoir un support écrit avec lequel ils pourront suivre le cours." />
  <meta name="generator" content="bookdown 0.29 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Analyse des Interactions de protéines | Tutoriel de Bioinformatique" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Ce document est recueil qui permettra aux étudiants d’avoir un support écrit avec lequel ils pourront suivre le cours." />
  <meta name="github-repo" content="kmezhoud/Bioinformatics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Analyse des Interactions de protéines | Tutoriel de Bioinformatique" />
  
  <meta name="twitter:description" content="Ce document est recueil qui permettra aux étudiants d’avoir un support écrit avec lequel ils pourront suivre le cours." />
  

<meta name="author" content="Karim Mezhoud" />


<meta name="date" content="2022-10-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="le-docking.html"/>
<link rel="next" href="autres.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/vis-9.1.0/vis-network.min.css" rel="stylesheet" />
<script src="libs/vis-9.1.0/vis-network.min.js"></script>
<script src="libs/visNetwork-binding-2.1.2/visNetwork.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Tutoriel de Bioinformatique</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction à la Bioinformatique</a></li>
<li class="chapter" data-level="2" data-path="les-bases-de-données.html"><a href="les-bases-de-données.html"><i class="fa fa-check"></i><b>2</b> Les Bases de Données</a></li>
<li class="chapter" data-level="3" data-path="cross.html"><a href="cross.html"><i class="fa fa-check"></i><b>3</b> Principes de l’Alignement de Séquences</a></li>
<li class="chapter" data-level="4" data-path="les-principes-de-prédiction-des-structures-3d-des-protéines.html"><a href="les-principes-de-prédiction-des-structures-3d-des-protéines.html"><i class="fa fa-check"></i><b>4</b> Les Principes de Prédiction des Structures 3D des protéines</a></li>
<li class="chapter" data-level="5" data-path="le-docking.html"><a href="le-docking.html"><i class="fa fa-check"></i><b>5</b> Le Docking</a></li>
<li class="chapter" data-level="6" data-path="analyse-des-interactions-de-protéines.html"><a href="analyse-des-interactions-de-protéines.html"><i class="fa fa-check"></i><b>6</b> Analyse des Interactions de protéines</a>
<ul>
<li class="chapter" data-level="6.1" data-path="analyse-des-interactions-de-protéines.html"><a href="analyse-des-interactions-de-protéines.html#cas-détude-identification-graphique-de-sous-réseaux-de-gènes-associés-aux-cancers-basé-sur-des-données-protéomiques-réduites"><i class="fa fa-check"></i><b>6.1</b> Cas d’étude: Identification Graphique de sous-réseaux de gènes associés aux cancers basé sur des données protéomiques réduites</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="analyse-des-interactions-de-protéines.html"><a href="analyse-des-interactions-de-protéines.html#tools-data"><i class="fa fa-check"></i><b>6.1.1</b> Tools &amp; Data</a></li>
<li class="chapter" data-level="6.1.2" data-path="analyse-des-interactions-de-protéines.html"><a href="analyse-des-interactions-de-protéines.html#data-computing"><i class="fa fa-check"></i><b>6.1.2</b> Data Computing</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="autres.html"><a href="autres.html"><i class="fa fa-check"></i><b>7</b> Autres</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://kmezhoud.netlify.app/" target="blank">Karim Mezhoud</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutoriel de Bioinformatique</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="analyse-des-interactions-de-protéines" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Analyse des Interactions de protéines<a href="analyse-des-interactions-de-protéines.html#analyse-des-interactions-de-protéines" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<ol style="list-style-type: decimal">
<li><p>Orale presentation</p></li>
<li><p>Include slides</p>
<ul>
<li>second level
<ul>
<li>third level</li>
</ul></li>
</ul></li>
</ol>
<div id="cas-détude-identification-graphique-de-sous-réseaux-de-gènes-associés-aux-cancers-basé-sur-des-données-protéomiques-réduites" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Cas d’étude: Identification Graphique de sous-réseaux de gènes associés aux cancers basé sur des données protéomiques réduites<a href="analyse-des-interactions-de-protéines.html#cas-détude-identification-graphique-de-sous-réseaux-de-gènes-associés-aux-cancers-basé-sur-des-données-protéomiques-réduites" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This tutorial describes the main steps used to construct an enriched network from a list of genes with their expression values. The network includes the fold change, the False discovery rate (FDR), the gene expression correlation, the most often involved biological processes, and the most likely corresponding diseases. All the main steps are illustrated by screen shots. The <a href="analyse-des-interactions-de-protéines.html#tools-data">Tools &amp; Data</a> section lists all the tools and data required. The “Data computing” section shows you how to compute the data. The Experimental procedure section lists briefly all the steps, and the Walk-through example section illustrates these steps. The final section lists all the software and packages that were tested during this study, but not selected for inclusion in this Note.</p>
<div id="tools-data" class="section level3 hasAnchor" number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> Tools &amp; Data<a href="analyse-des-interactions-de-protéines.html#tools-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For this tutorial you will need:</p>
<ol style="list-style-type: decimal">
<li>Bioinformatic tools</li>
</ol>
<p>Cytoscape <span class="citation">(<a href="#ref-Shannon2003" role="doc-biblioref">Shannon et al. 2003</a>; <a href="#ref-Cline2007" role="doc-biblioref">Cline et al. 2007</a>)</span> software implemented by Reactome FIs <span class="citation">(<a href="#ref-Wu2010" role="doc-biblioref">Wu, Feng, and Stein 2010</a>)</span>, Advanced Network merge, NetworkAnalyzer, and ClueGO <span class="citation">(<a href="#ref-Bindea2009" role="doc-biblioref">Bindea et al. 2009</a>)</span> Apps.</p>
<ol start="2" style="list-style-type: decimal">
<li>Experimental Data</li>
</ol>
<p>To find out more about the file format, download the corresponding supplemental file indicated in parentheses.</p>
<ul>
<li><p>The list of modulated protein saved in one column in text file (only_modulated_symbol.txt).
Convert the protein ID on gene symbol (HGNC or HUGO symbol) using ID Mapping service, such as:</p>
<ul>
<li><a href="https://www.ensembl.org/info/data/biomart/index.html">BioMart</a></li>
<li><a href="http://www.uniprot.org/">Uniprot</a></li>
<li><a href="https://cran.r-project.org/web/packages/IDConverter/index.html">Idconverter</a> <span class="citation">(<a href="#ref-Alibes2007" role="doc-biblioref">Alibés et al. 2007</a>)</span></li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/gage.html">GAGE</a> <span class="citation">(<a href="#ref-Luo2009" role="doc-biblioref">Luo et al. 2009</a>)</span> with the <code>eg2sym</code> function.</li>
</ul></li>
<li><p>The protein expression values. The following example of gene expression is divided into two groups and 3 replicates. (proteinEXP.txt)</p></li>
<li><p>Node Attributes</p>
<ul>
<li>Fold change of the protein identified (Fold_Change.NA)</li>
<li>False discovery rate of the z score or some other statistical test (FDR.NA)</li>
<li>Correlated protein expression by statistical analysis (Correlation_Network.sif)</li>
<li>Cancer-protein network (Cancer_Network.sif)</li>
</ul></li>
<li><p>Edge attributes</p>
<ul>
<li>The correlation rate of Protein expression (Correlation+Cancer.EA)</li>
<li>The gene expression profile overlap between our experiments and clinical data from the <a href="(http://www.cbioportal.org/)">cBio Cancer Genomics Portal</a> <span class="citation">(<a href="#ref-Cerami2012" role="doc-biblioref">Cerami et al. 2012</a>)</span> (Correlation+Cancer.EA)</li>
</ul></li>
</ul>
</div>
<div id="data-computing" class="section level3 hasAnchor" number="6.1.2">
<h3><span class="header-section-number">6.1.2</span> Data Computing<a href="analyse-des-interactions-de-protéines.html#data-computing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="gene-list" class="section level4 hasAnchor" number="6.1.2.1">
<h4><span class="header-section-number">6.1.2.1</span> Gene list<a href="analyse-des-interactions-de-protéines.html#gene-list" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We assume that the protein list with quantitative analysis has already been compiled . Supplemental <strong>proteinEXP.txt</strong> file gives an example of the quantitative data which will be used.</p>
<p>Pre-process the expression data into a format suitable for network analysis, clean the data by removing obvious outlier samples or excessive numbers of missing entries, and then load the data. The gene list is selected after statistical analysis.
The Fold change and False discovery rate are then computed and saved in the corresponding files: <strong>Fold_Change.NA</strong> and <strong>FDR.NA</strong>.</p>
</div>
<div id="multiple-correlation-analysis-of-protein-expression" class="section level4 hasAnchor" number="6.1.2.2">
<h4><span class="header-section-number">6.1.2.2</span> Multiple Correlation analysis of protein expression<a href="analyse-des-interactions-de-protéines.html#multiple-correlation-analysis-of-protein-expression" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb1"><pre class="sourceCode numberSource r numberLines"><code class="sourceCode r"><span id="cb1-1"><a href="analyse-des-interactions-de-protéines.html#cb1-1"></a><span class="fu">library</span>(visNetwork)</span>
<span id="cb1-2"><a href="analyse-des-interactions-de-protéines.html#cb1-2"></a>nodes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb1-3"><a href="analyse-des-interactions-de-protéines.html#cb1-3"></a>                    <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">&quot;Node&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>),</span>
<span id="cb1-4"><a href="analyse-des-interactions-de-protéines.html#cb1-4"></a>                    <span class="at">value =</span> <span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">5</span>,<span class="dv">3</span>,<span class="dv">7</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="fl">2.6</span>,<span class="dv">2</span>,<span class="dv">9</span>,<span class="dv">2</span>),</span>
<span id="cb1-5"><a href="analyse-des-interactions-de-protéines.html#cb1-5"></a>                    <span class="at">shape =</span><span class="st">&quot;circle&quot;</span>,</span>
<span id="cb1-6"><a href="analyse-des-interactions-de-protéines.html#cb1-6"></a>                    <span class="at">group=</span> <span class="fu">c</span>(<span class="st">&quot;Gr1&quot;</span>,<span class="st">&quot;Gr3&quot;</span>),</span>
<span id="cb1-7"><a href="analyse-des-interactions-de-protéines.html#cb1-7"></a>                    <span class="at">color=</span> <span class="fu">c</span>(<span class="st">&quot;#DF0101&quot;</span>, <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb1-8"><a href="analyse-des-interactions-de-protéines.html#cb1-8"></a>)</span>
<span id="cb1-9"><a href="analyse-des-interactions-de-protéines.html#cb1-9"></a></span>
<span id="cb1-10"><a href="analyse-des-interactions-de-protéines.html#cb1-10"></a>edges <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">from =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">3</span>,<span class="dv">7</span>,<span class="dv">2</span>,<span class="dv">9</span>,<span class="dv">10</span>,<span class="dv">8</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">6</span>),</span>
<span id="cb1-11"><a href="analyse-des-interactions-de-protéines.html#cb1-11"></a>                    <span class="at">to =</span> <span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">7</span>,<span class="dv">4</span>,<span class="dv">9</span>) )</span>
<span id="cb1-12"><a href="analyse-des-interactions-de-protéines.html#cb1-12"></a></span>
<span id="cb1-13"><a href="analyse-des-interactions-de-protéines.html#cb1-13"></a></span>
<span id="cb1-14"><a href="analyse-des-interactions-de-protéines.html#cb1-14"></a></span>
<span id="cb1-15"><a href="analyse-des-interactions-de-protéines.html#cb1-15"></a>graphe <span class="ot">&lt;-</span> visNetwork<span class="sc">::</span><span class="fu">visNetwork</span>(nodes, edges, <span class="at">width =</span> <span class="st">&quot;100%&quot;</span>)</span>
<span id="cb1-16"><a href="analyse-des-interactions-de-protéines.html#cb1-16"></a></span>
<span id="cb1-17"><a href="analyse-des-interactions-de-protéines.html#cb1-17"></a>visNetwork<span class="sc">::</span><span class="fu">visOptions</span>(<span class="at">graph=</span> graphe, <span class="at">manipulation =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-18"><a href="analyse-des-interactions-de-protéines.html#cb1-18"></a>                       <span class="at">selectedBy =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb1-19"><a href="analyse-des-interactions-de-protéines.html#cb1-19"></a>                       <span class="at">highlightNearest =</span> <span class="cn">TRUE</span> )</span></code></pre></div>
<div id="htmlwidget-36664edc69a1212deeb2" style="width:100%;height:480px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-36664edc69a1212deeb2">{"x":{"nodes":{"id":[1,2,3,4,5,6,7,8,9,10],"label":["Node 1","Node 2","Node 3","Node 4","Node 5","Node 6","Node 7","Node 8","Node 9","Node 10"],"value":[10,5,3,7,2,3,2.6,2,9,2],"shape":["circle","circle","circle","circle","circle","circle","circle","circle","circle","circle"],"group":["Gr1","Gr3","Gr1","Gr3","Gr1","Gr3","Gr1","Gr3","Gr1","Gr3"],"color":["#DF0101","blue","#DF0101","blue","#DF0101","blue","#DF0101","blue","#DF0101","blue"]},"edges":{"from":[1,5,3,7,2,9,10,8,3,3,6],"to":[5,3,5,1,5,1,2,3,7,4,9]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot"},"manipulation":{"enabled":true}},"groups":["Gr1","Gr3"],"width":"100%","height":null,"idselection":{"enabled":false,"style":"width: 150px; height: 26px","useLabels":true,"main":"Select by id"},"byselection":{"enabled":true,"style":"width: 150px; height: 26px","multiple":false,"hideColor":"rgba(200,200,200,0.5)","highlight":false,"variable":"group","main":"Select by group","values":["Gr1","Gr3"]},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","opts_manipulation":{"datacss":"table.legend_table {\n  font-size: 11px;\n  border-width:1px;\n  border-color:#d3d3d3;\n  border-style:solid;\n}\ntable.legend_table td {\n  border-width:1px;\n  border-color:#d3d3d3;\n  border-style:solid;\n  padding: 2px;\n}\ndiv.table_content {\n  width:80px;\n  text-align:center;\n}\ndiv.table_description {\n  width:100px;\n}\n\n.operation {\n  font-size:20px;\n}\n\n.network-popUp {\n  display:none;\n  z-index:299;\n  width:250px;\n  /*height:150px;*/\n  background-color: #f9f9f9;\n  border-style:solid;\n  border-width:1px;\n  border-color: #0d0d0d;\n  padding:10px;\n  text-align: center;\n  position:fixed;\n  top:50%;  \n  left:50%;  \n  margin:-100px 0 0 -100px;  \n\n}","addNodeCols":["id","label"],"editNodeCols":["id","label"],"tab_add_node":"<span id=\"addnode-operation\" class = \"operation\">node<\/span> <br><table style=\"margin:auto;\"><tr><td>id<\/td><td><input id=\"addnode-id\"  type= \"text\" value=\"new value\"><\/td><\/tr><tr><td>label<\/td><td><input id=\"addnode-label\"  type= \"text\" value=\"new value\"><\/td><\/tr><\/table><input type=\"button\" value=\"save\" id=\"addnode-saveButton\"><\/button><input type=\"button\" value=\"cancel\" id=\"addnode-cancelButton\"><\/button>","tab_edit_node":"<span id=\"editnode-operation\" class = \"operation\">node<\/span> <br><table style=\"margin:auto;\"><tr><td>id<\/td><td><input id=\"editnode-id\"  type= \"text\" value=\"new value\"><\/td><\/tr><tr><td>label<\/td><td><input id=\"editnode-label\"  type= \"text\" value=\"new value\"><\/td><\/tr><\/table><input type=\"button\" value=\"save\" id=\"editnode-saveButton\"><\/button><input type=\"button\" value=\"cancel\" id=\"editnode-cancelButton\"><\/button>"},"highlight":{"enabled":true,"hoverNearest":false,"degree":1,"algorithm":"all","hideColor":"rgba(200,200,200,0.5)","labelOnly":true},"collapse":{"enabled":false,"fit":false,"resetHighlight":true,"clusterOptions":null,"keepCoord":true,"labelSuffix":"(cluster)"}},"evals":[],"jsHooks":[]}</script>
<!-- # Blocks -->
<!-- ## Equations -->
<!-- Here is an equation. -->
<!-- \begin{equation}  -->
<!--   f\left(k\right) = \binom{n}{k} p^k\left(1-p\right)^{n-k} -->
<!--   (\#eq:binom) -->
<!-- \end{equation}  -->
<!-- You may refer to using `\@ref(eq:binom)`, like see Equation \@ref(eq:binom). -->
<!-- ## Theorems and proofs -->
<!-- Labeled theorems can be referenced in text using `\@ref(thm:tri)`, for example, check out this smart theorem \@ref(thm:tri). -->
<!-- ::: {.theorem #tri} -->
<!-- For a right triangle, if $c$ denotes the *length* of the hypotenuse -->
<!-- and $a$ and $b$ denote the lengths of the **other** two sides, we have -->
<!-- $$a^2 + b^2 = c^2$$ -->
<!-- ::: -->
<!-- Read more here <https://bookdown.org/yihui/bookdown/markdown-extensions-by-bookdown.html>. -->
<!-- ## Callout blocks -->
<!-- The R Markdown Cookbook provides more help on how to use custom blocks to design your own callouts: https://bookdown.org/yihui/rmarkdown-cookbook/custom-blocks.html -->

</div>
</div>
</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Alibes2007" class="csl-entry">
Alibés, Andreu, Patricio Yankilevich, Andrés Cañada, and Ramón Díaz-Uriarte. 2007. <span>“IDconverter and IDClight: Conversion and Annotation of Gene and Protein IDs.”</span> <em>BMC Bioinformatics</em> 8 (January): 9. <a href="https://doi.org/10.1186/1471-2105-8-9">https://doi.org/10.1186/1471-2105-8-9</a>.
</div>
<div id="ref-Bindea2009" class="csl-entry">
Bindea, Gabriela, Bernhard Mlecnik, Hubert Hackl, Pornpimol Charoentong, Marie Tosolini, Amos Kirilovsky, Wolf-Herman Fridman, Franck Pagès, Zlatko Trajanoski, and Jérôme Galon. 2009. <span>“ClueGO: A Cytoscape Plug-in to Decipher Functionally Grouped Gene Ontology and Pathway Annotation Networks.”</span> <em>Bioinformatics (Oxford, England)</em> 25 (April): 1091–93. <a href="https://doi.org/10.1093/bioinformatics/btp101">https://doi.org/10.1093/bioinformatics/btp101</a>.
</div>
<div id="ref-Cerami2012" class="csl-entry">
Cerami, Ethan, Jianjiong Gao, Ugur Dogrusoz, Benjamin E. Gross, Selcuk Onur Sumer, Bülent Arman Aksoy, Anders Jacobsen, et al. 2012. <span>“The cBio Cancer Genomics Portal: An Open Platform for Exploring Multidimensional Cancer Genomics Data.”</span> <em>Cancer Discovery</em> 2 (May): 401–4. <a href="https://doi.org/10.1158/2159-8290.CD-12-0095">https://doi.org/10.1158/2159-8290.CD-12-0095</a>.
</div>
<div id="ref-Cline2007" class="csl-entry">
Cline, Melissa S., Michael Smoot, Ethan Cerami, Allan Kuchinsky, Nerius Landys, Chris Workman, Rowan Christmas, et al. 2007. <span>“Integration of Biological Networks and Gene Expression Data Using Cytoscape.”</span> <em>Nature Protocols</em> 2: 2366–82. <a href="https://doi.org/10.1038/nprot.2007.324">https://doi.org/10.1038/nprot.2007.324</a>.
</div>
<div id="ref-Luo2009" class="csl-entry">
Luo, Weijun, Michael S. Friedman, Kerby Shedden, Kurt D. Hankenson, and Peter J. Woolf. 2009. <span>“GAGE: Generally Applicable Gene Set Enrichment for Pathway Analysis.”</span> <em>BMC Bioinformatics</em> 10 (May): 161. <a href="https://doi.org/10.1186/1471-2105-10-161">https://doi.org/10.1186/1471-2105-10-161</a>.
</div>
<div id="ref-Shannon2003" class="csl-entry">
Shannon, Paul, Andrew Markiel, Owen Ozier, Nitin S. Baliga, Jonathan T. Wang, Daniel Ramage, Nada Amin, Benno Schwikowski, and Trey Ideker. 2003. <span>“Cytoscape: A Software Environment for Integrated Models of Biomolecular Interaction Networks.”</span> <em>Genome Research</em> 13 (November): 2498–2504. <a href="https://doi.org/10.1101/gr.1239303">https://doi.org/10.1101/gr.1239303</a>.
</div>
<div id="ref-Wu2010" class="csl-entry">
Wu, Guanming, Xin Feng, and Lincoln Stein. 2010. <span>“A Human Functional Protein Interaction Network and Its Application to Cancer Data Analysis.”</span> <em>Genome Biology</em> 11: R53. <a href="https://doi.org/10.1186/gb-2010-11-5-r53">https://doi.org/10.1186/gb-2010-11-5-r53</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="le-docking.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="autres.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": null,
"edit": {
"link": "https://github.com/kmezhoud/Bioinformatics/edit/main/05-interactomics.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
